modules/tensor_mechanics/include/materials/ADComputeEigenstrainBase.h

```diff
@@ -0,0 +1,74 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef ADADCOMPUTEEIGENSTRAINBASE_H
+#define ADADCOMPUTEEIGENSTRAINBASE_H
+
+#include "ADMaterial.h"
+
+#define adDeclareRestartableData this->template declareRestartableData
```

Comment:
@lindsayad I wasn't sure if this is the correct way to do this?

Reply:
This does follow our heuristic. However, I think it should be put at the top of `Restartable.h` after the forward declarations, since that is the file where the template method is declared/defined and other developers might want that macro available in the future.